magefile.go

```diff
@@ -689,9 +689,19 @@ func packageAgent(requiredPackages []string, packagingFn func()) {
 				}
 			}
 		} else {
+			// This shouldn't be needed as a special append() here, but until our artiface APIs have released shipper packages,
+			// we need to make sure the package command doesn't try to download a shipper tarball and blow up
+			packedBeats = append(packedBeats, "elastic-agent-shipper")
 			// build from local repo, will assume beats repo is located on the same root level
+			fmt.Println(">>> Building from local repo")
 			for _, b := range packedBeats {
-				pwd, err := filepath.Abs(filepath.Join("../beats/x-pack", b))
+				var pwd string
+				var err error
+				if !strings.Contains(b, "beat") {
```

Comment:
i'd rather have this explicitely configured e.g changing packedBeats to a struct or a map where key is a beat name and value is `IsXpack`